modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/upgrade/v13_0_1/test/ReturnsManagerRoleUpgradeProcessTest.java

```diff
@@ -0,0 +1,187 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Crescenzo Rega
+ */
+@RunWith(Arquillian.class)
+public class ReturnsManagerRoleUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		_removeResourcePermissions();
+
+		_runUpgrade();
+
+		Assert.assertTrue(
+			ListUtil.exists(
+				_resourcePermissionLocalService.getRoles(
+					_group.getCompanyId(), CommercePortletKeys.COMMERCE_PAYMENT,
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(_group.getCompanyId()),
+					ActionKeys.ACCESS_IN_CONTROL_PANEL),
+				role -> StringUtil.equals(
+					role.getName(),
+					AccountRoleConstants.ROLE_NAME_RETURNS_MANAGER)));
+		Assert.assertTrue(
+			ListUtil.exists(
+				_resourcePermissionLocalService.getRoles(
+					_group.getCompanyId(), CommerceChannel.class.getName(),
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(_group.getCompanyId()), ActionKeys.UPDATE),
+				role -> StringUtil.equals(
+					role.getName(),
+					AccountRoleConstants.ROLE_NAME_RETURNS_MANAGER)));
+		Assert.assertTrue(
+			ListUtil.exists(
+				_resourcePermissionLocalService.getRoles(
+					_group.getCompanyId(), CommerceChannel.class.getName(),
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(_group.getCompanyId()), ActionKeys.VIEW),
+				role -> StringUtil.equals(
+					role.getName(),
+					AccountRoleConstants.ROLE_NAME_RETURNS_MANAGER)));
+		Assert.assertTrue(
+			ListUtil.exists(
+				_resourcePermissionLocalService.getRoles(
+					_group.getCompanyId(), "com.liferay.commerce.currency",
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(_group.getCompanyId()),
+					CommerceCurrencyActionKeys.MANAGE_COMMERCE_CURRENCIES),
+				role -> StringUtil.equals(
+					role.getName(),
+					AccountRoleConstants.ROLE_NAME_RETURNS_MANAGER)));
+		Assert.assertTrue(
+			ListUtil.exists(
+				_resourcePermissionLocalService.getRoles(
+					_group.getCompanyId(), "com.liferay.commerce.payment",
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(_group.getCompanyId()),
+					CommercePaymentEntryActionKeys.ADD_REFUND),
+				role -> StringUtil.equals(
+					role.getName(),
+					AccountRoleConstants.ROLE_NAME_RETURNS_MANAGER)));
+	}
+
+	private void _removeResourcePermissions() throws Exception {
+		Role role = _roleLocalService.fetchRole(
+			_group.getCompanyId(),
+			AccountRoleConstants.ROLE_NAME_RETURNS_MANAGER);
+
+		_resourcePermissionLocalService.removeResourcePermission(
+			_group.getCompanyId(), CommercePortletKeys.COMMERCE_PAYMENT,
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(_group.getCompanyId()), role.getRoleId(),
+			ActionKeys.ACCESS_IN_CONTROL_PANEL);
+
```

Comment:
I don't know if the empty lines between the ...removeResourcePermission are needed or not. Could you please try removing them and see if sf complains?